boards/tutorials/nrf52840dk-process-control/src/main.rs

```diff
@@ -0,0 +1,641 @@
+
+    // Create a virtual flash user for dynamic binary storage
+    let virtual_flash_dbs = components::flash::FlashUserComponent::new(mux_flash).finalize(
+        components::flash_user_component_static!(nrf52840::nvmc::Nvmc),
+    );
+
+    // Create a virtual flash user for nonvolatile
+    let virtual_flash_nvm = components::flash::FlashUserComponent::new(mux_flash).finalize(
+        components::flash_user_component_static!(nrf52840::nvmc::Nvmc),
+    );
+
+    //--------------------------------------------------------------------------
+    // Credential Checking
+    //--------------------------------------------------------------------------
+
+    // Create the credential checker.
+    let checking_policy = components::appid::checker_null::AppCheckerNullComponent::new()
+        .finalize(components::app_checker_null_component_static!());
+
+    // Create the AppID assigner.
+    let assigner = components::appid::assigner_tbf::AppIdAssignerTbfHeaderComponent::new()
+        .finalize(components::appid_assigner_tbf_header_component_static!());
+
+    // Create the process checking machine.
+    let checker = components::appid::checker::ProcessCheckerMachineComponent::new(checking_policy)
+        .finalize(components::process_checker_machine_component_static!());
+
+    //--------------------------------------------------------------------------
+    // STORAGE PERMISSIONS
+    //--------------------------------------------------------------------------
+
+    let storage_permissions_policy =
+        components::storage_permissions::null::StoragePermissionsNullComponent::new().finalize(
+            components::storage_permissions_null_component_static!(
+                nrf52840::chip::NRF52<Nrf52840DefaultPeripherals>,
+                kernel::process::ProcessStandardDebugFull,
+            ),
+        );
+
+    // Create and start the asynchronous process loader.
+    let loader = components::loader::sequential::ProcessLoaderSequentialComponent::new(
+        checker,
+        &mut *addr_of_mut!(PROCESSES),
+        board_kernel,
+        chip,
+        &FAULT_RESPONSE,
+        assigner,
+        storage_permissions_policy,
+    )
+    .finalize(components::process_loader_sequential_component_static!(
+        nrf52840::chip::NRF52<Nrf52840DefaultPeripherals>,
+        kernel::process::ProcessStandardDebugFull,
+        NUM_PROCS
+    ));
+
+    //--------------------------------------------------------------------------
+    // Dynamic App Loading
+    //--------------------------------------------------------------------------
+
+    // Create the dynamic binary flasher.
+    let dynamic_binary_storage =
+        components::dynamic_binary_storage::SequentialBinaryStorageComponent::new(
+            virtual_flash_dbs,
+            loader,
+        )
+        .finalize(components::sequential_binary_storage_component_static!(
+            capsules_core::virtualizers::virtual_flash::FlashUser<'static, nrf52840::nvmc::Nvmc>,
```

Comment:
Can you create a `type` for this as well?

We are going to need to have docs on how to add this to other boards, and it would be great to have it as copy-and-paste-able as possible.